src/OrleansSQLUtils/Storage/DbStoredQueries.cs

```diff
@@ -0,0 +1,406 @@
+                    Period = TimeSpan.FromMilliseconds(record.GetValue<int>("Period")),
+                    ETag = GetVersion(record).ToString()
+                };
+            }
+            return null;
+        }
+
+        internal static Tuple<MembershipEntry, int> GetMembershipEntry(IDataRecord record)
+        {
+            //TODO: This is a bit of hack way to check in the current version if there's membership data or not, but if there's a start time, there's member.            
+            DateTime? startTime = record.GetValueOrDefault<DateTime?>("StartTime");
+            MembershipEntry entry = null;
+            if (startTime.HasValue)
+            {
+                entry = new MembershipEntry
+                {
+                    SiloAddress = GetSiloAddress(record, "Port"),
+                    HostName = record.GetValue<string>("HostName"),
+                    Status = record.GetValue<SiloStatus>("Status"),
+                    ProxyPort = record.GetValue<int>("ProxyPort"),
+                    StartTime = startTime.Value,
+                    IAmAliveTime = record.GetValue<DateTime>("IAmAliveTime")
+                };
+
+                string suspectingSilos = record.GetValueOrDefault<string>("SuspectTimes");
+                if (!string.IsNullOrWhiteSpace(suspectingSilos))
+                {
+                    entry.SuspectTimes = new List<Tuple<SiloAddress, DateTime>>();
+                    entry.SuspectTimes.AddRange(suspectingSilos.Split('|').Select(s =>
+                    {
+                        var split = s.Split(',');
+                        return new Tuple<SiloAddress, DateTime>(SiloAddress.FromParsableString(split[0]),
+                            TraceLogger.ParseDate(split[1]));
+                    }));
+                }
+            }
+
+            return Tuple.Create(entry, GetVersion(record));
+        }
+
+        internal static int GetVersion(IDataRecord record)
+        {
+            return Convert.ToInt32(record.GetValue<object>("Version"));
+        }
+
+        internal static Uri GetGatewayUri(IDataRecord record)
+        {
+            return GetSiloAddress(record, "ProxyPort").ToGatewayUri();
+        }
+
+        private static SiloAddress GetSiloAddress(IDataRecord record, string portName)
+        {
+            int port = record.GetValue<int>(portName);
+            int generation = record.GetValue<int>("Generation");
+            string address = record.GetValue<string>("Address");
+            var siloAddress = SiloAddress.New(new IPEndPoint(IPAddress.Parse(address), port), generation);
+            return siloAddress;
+        }
+
+        internal static bool GetSingleBooleanValue(IDataRecord record)
+        {
+            if (record.FieldCount != 1) throw new ArgumentException("expected a single column", nameof(record));
+            return Convert.ToBoolean(record.GetValue(0));
+        }
+
+#pragma warning restore 0649
+        internal class Params
```

Comment:
I wondered about the naming. Maybe it'd be better to just name everything C# style, the parameters too. So we'd have the same parameter and column names, IMO will be easier to follow. The `DbStoredQueries` class is now the only one that "knows" about parameters, queries and column names. the nested `Params` class is responsible for all parameter definitions, and since a lot of them are the same in many queries, I've put them together and by that - added a bit of more indirection, but we gained separation of concerns.